include/pybind11/chrono.h

```diff
@@ -0,0 +1,177 @@
+/*
+    pybind11/chrono.h: Transparent conversion between std::chrono and python's datetime
+
+    Copyright (c) 2016 Trent Houliston <trent@houliston.me> and
+                       Wenzel Jakob <wenzel.jakob@epfl.ch>
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+#pragma once
+
+#include "pybind11.h"
+#include <cmath>
+#include <chrono>
+#include <datetime.h>
+
+NAMESPACE_BEGIN(pybind11)
+NAMESPACE_BEGIN(detail)
+
+template <typename Rep, typename Period> class type_caster<std::chrono::duration<Rep, Period>> {
+public:
+    typedef std::chrono::duration<Rep, Period> type;
+    typedef std::chrono::duration<std::chrono::hours::rep, std::ratio<86400>> days;
```

Comment:
`std::chrono::hours::rep`, in theory, might not be big enough to hold python's possible days range (which the API says can be in the range of -999,999,999 to 999,999,999).  Perhaps change this to `std::chrono::duration<int_fast32_t, std::ratio<86400>>`, which I think is a little clearer than the above anyway.